src/vs/workbench/services/configurationResolver/test/electron-sandbox/configurationResolverService.test.ts

```diff
@@ -188,7 +213,7 @@ suite('Configuration Resolver Service', () => {
 	});
 
 	test('disallows nested keys (#77289)', async () => {
-		assert.strictEqual(await configurationResolverService!.resolveAsync(workspace, '${env:key1} ${env:key1${env:key2}}'), 'Value for key1 ${env:key1${env:key2}}');
+		assert.strictEqual(await configurationResolverService!.resolveAsync(workspace, '${env:key1} ${env:key1${env:key2}}'), 'Value for key1 ');
```

Comment:
This seems suspicious. Does this mean that unresolved keys are now empty string?

Reply:
Just nested keys, this now actually looks up the environment variable `key1${env:key2}`, which is undefined/empty.

This test was added in response to https://github.com/microsoft/vscode/issues/77289. Arguable what we should do here, before we just didn't resolve it. I feel like the approach of treating it as a variable is a bit better, and is easy to do now that we use an iterative parser rather than regex.

Reply:
If it's just nested keys, then that seems less likely to cause unexpected behavior for folks 👍